google-cloud-spanner/lib/google/cloud/spanner/database.rb

```diff
@@ -443,14 +451,38 @@ def database_operations filter: nil, page_size: nil
         #     backup = job.backup
         #   end
         #
-        def create_backup backup_id, expire_time, version_time: nil
+        # @example Create backup with encryption config
+        #   require "google/cloud/spanner"
+        #
+        #   spanner = Google::Cloud::Spanner.new
+        #   database = spanner.database "my-instance", "my-database"
+        #
+        #   kms_key_name = "projects/<project>/locations/<location>/keyRings/<key_ring>/cryptoKeys/<kms_key_name>"
+        #   encryption_config = { kms_key_name: kms_key_name }
+        #   job = database.create_backup "my-backup",
+        #                                Time.now + 36000,
+        #                                encryption_config: encryption_config
+        #
+        #   job.done? #=> false
+        #   job.reload! # API call
+        #   job.done? #=> true
+        #
+        #   if job.error?
+        #     status = job.error
+        #   else
+        #     backup = job.backup
+        #   end
+        #
+        def create_backup backup_id, expire_time,
+                          version_time: nil, encryption_config: nil
```

Comment:
The `encryption_config` of a `backup` has different options than a `database`'s encryption config. See https://github.com/googleapis/googleapis/blob/master/google/spanner/admin/database/v1/backup.proto#L157-L158 and https://github.com/googleapis/googleapis/blob/master/google/spanner/admin/database/v1/backup.proto#L425-L460. This means that a backup encryption config can be one of the following:

- [CustomerManagedEncryption](https://github.com/googleapis/googleapis/blob/master/google/spanner/admin/database/v1/backup.proto#L444): We need to specify a `kms_key_name` when this option is used.
- [GoogleDefaultEncryption](https://github.com/googleapis/googleapis/blob/master/google/spanner/admin/database/v1/backup.proto#L440): no key should be specified.
- [UseDatabaseEncryption](https://github.com/googleapis/googleapis/blob/master/google/spanner/admin/database/v1/backup.proto#L437): no key should be specified.

Reply:
ok, I will update it.